<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>LineProximityGrouping_Widget</class>
 <widget class="QWidget" name="LineProximityGrouping_Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>300</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Line Proximity Grouping</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="parametersGroupBox">
     <property name="title">
      <string>Grouping Parameters</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="distanceThresholdLabel">
        <property name="text">
         <string>Distance Threshold:</string>
        </property>
        <property name="toolTip">
         <string>Maximum distance between lines to be considered for the same group</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QDoubleSpinBox" name="distanceThresholdSpinBox">
        <property name="toolTip">
         <string>Maximum distance between lines to be considered for the same group</string>
        </property>
        <property name="suffix">
         <string> pixels</string>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.010000000000000</double>
        </property>
        <property name="maximum">
         <double>1000.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>1.000000000000000</double>
        </property>
        <property name="value">
         <double>10.000000000000000</double>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="positionLabel">
        <property name="text">
         <string>Position Along Line:</string>
        </property>
        <property name="toolTip">
         <string>Fractional position along each line where distance is measured (0.0 = start, 1.0 = end)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="positionSpinBox">
        <property name="toolTip">
         <string>Fractional position along each line where distance is measured (0.0 = start, 1.0 = end)</string>
        </property>
        <property name="decimals">
         <number>2</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>1.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
        <property name="value">
         <double>0.500000000000000</double>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="outlierGroupBox">
     <property name="title">
      <string>Outlier Handling</string>
     </property>
     <layout class="QVBoxLayout" name="outlierLayout">
      <item>
       <widget class="QCheckBox" name="createNewGroupCheckBox">
        <property name="text">
         <string>Create new group for outliers</string>
        </property>
        <property name="toolTip">
         <string>If enabled, lines that don't fit any existing group will be placed in a new group</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <layout class="QFormLayout" name="newGroupLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="newGroupNameLabel">
          <property name="text">
           <string>New Group Name:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="newGroupNameLineEdit">
          <property name="toolTip">
           <string>Name for the new group that will contain outlier lines</string>
          </property>
          <property name="text">
           <string>Ungrouped Lines</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="infoLabel">
     <property name="text">
      <string>This operation groups ungrouped lines with existing groups based on proximity at a specified position along each line.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
     <property name="styleSheet">
      <string>QLabel { color: #666; font-style: italic; margin: 10px; }</string>
     </property>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>createNewGroupCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>newGroupNameLabel</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>100</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>createNewGroupCheckBox</sender>
   <signal>toggled(bool)</signal>
   <receiver>newGroupNameLineEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>200</x>
     <y>180</y>
    </hint>
    <hint type="destinationlabel">
     <x>250</x>
     <y>210</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>